service/common/src/main/java/org/apache/polaris/service/catalog/generic/GenericTableCatalogHandler.java

```diff
@@ -94,7 +99,14 @@ public LoadGenericTableResponse createGenericTable(
             createdEntity.getFormat(),
             createdEntity.getDoc(),
             createdEntity.getPropertiesAsMap());
-
+    if (XTableConversionUtils.requiresConversion(callContext, properties)) {
+      XTableConverter converter =
+          new XTableConverter(XTableConversionUtils.getHostUrl(callContext));
+      RunSyncResponse runSyncResponse = converter.execute(createdEntity);
+      createdTable
+          .getProperties()
+          .put(TARGET_FORMAT_METADATA_PATH_KEY, runSyncResponse.getTargetMetadataPath());
+    }
```

Comment:
instead of duplicating this below, should this go into some common place?

Reply:
I can try refactoring this.